src/researchhub_document/views/researchhub_post_views.py

```diff
@@ -165,6 +207,10 @@ def update_existing_researchhub_posts(self, request):
             document_types=['all', 'posts']
         )
 
+        if assign_doi:
+            charge_doi_fee(created_by, rh_post)
```

Comment:
I think we should flip the operation and charge only if DOI was assigned.

Also, are we handling error on the FE in case crossref fails?

Reply:
Good point, I'll add the check and error handling.

Reply:
I added some test as well.

Reply:
I added some tests as well.